### ISSUE TITLE
FR: Custom MTU value for aws_network_interface

### ISSUE THREAD

**Issue by weslambert** [terraform-providers/terraform-provider-aws#12763]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I would like to request the option (if possible) to specify an MTU for an `aws_network_interface` resource.  I see the option is available for a [direct-connect interface](https://www.terraform.io/docs/providers/aws/r/dx_private_virtual_interface.html), but not for this type (used for EC2, etc).  Is this possible, or would this have to be done at the OS-level of the EC2 instance?

### New or Affected Resource(s)

`aws_network_interface`

### Potential Terraform Configuration

```hcl
resource "aws_network_interface" "mynetworkinterface" {
    subnet_id        = aws_subnet.default.id
    private_ips      = ["172.16.163.20"]
    security_groups  = [aws_security_group.mysecuritygroup.id]
    mtu              = <MTU value>
}
```

### References

None.

**Comment by DrFaust92** [comment]
I don't see that it's an option in the aws sdk

**Comment by weslambert** [comment]
So, I'm guessing we can only set via OS-level config, then?

**Comment by DrFaust92** [comment]
seems like it based on https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/network_mtu.html

**Comment by weslambert** [comment]
Right, in my case, I would like to deploy an instance with an MTU that is already configured, so it can be used as a mirror target for VPC mirroring (so that I can specify something other than 1500/9001, as VXLAN, etc adds 54 bytes).  I'm assuming the only way to do this in a more streamlined way would be to use the file provisioner/remote-exec.
